pkg/clusteragent/clusterchecks/dangling_config.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build clusterchecks
+
+package clusterchecks
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+)
+
+type danglingConfigWrapper struct {
+	config           integration.Config
+	timeCreated      time.Time
+	unscheduledCheck bool
+}
+
+// createDanglingConfig creates a new danglingConfigWrapper
+// This is used to keep track of the lifecycle of a dangling config
+func createDanglingConfig(config integration.Config) *danglingConfigWrapper {
```

Comment:
Is it better to pass a pointer to config here to avoid extra copy?

Reply:
The original structure did not have a pointer to the `integration.Config` so I kept the same design. All the other methods that pass around the config don't use pointers so there would be a bit of referencing and dereferencing occurring. Hence, I'm not sure if the pointer is needed here. Lmk what you think.